filesystem/squashfs_superblock.ksy

```diff
@@ -0,0 +1,112 @@
+meta:
+  id: squashfs_superblock
+  title: SquashFS superblock
+  file-extension:
+    - squashfs
+    - snap
```

Comment:
What about the `.sfs` extension mentioned at [http://fileformats.archiveteam.org/wiki/Squashfs](http://fileformats.archiveteam.org/wiki/Squashfs#:~:text=Extension(s)) and [https://fileinfo.com/extension/squashfs](https://fileinfo.com/extension/squashfs#:~:text=NOTE%3A%20Squashfs%20files%20more%20commonly,SFS%20extension.)?

Reply:
GitHub identifies `.sfs` as some audio format for games. Must be "sound font" or something.

https://github.com/search?p=5&q=extension%3Asfs&type=Code

In addition I haven't seen such usage. Neither it is present in spec or recommended.

Reply:
And what about `.sqfs`? It's used in

- https://github.com/AgentD/squashfs-tools-ng/blob/b76eae1/tests/corpus/cantrbry.sha512#L1
- https://boxmatrix.info/wiki/Property:modfw.sqfs
- https://www.plop.at/en/ploplinux/buildtools/full.html
- https://keybase.pub/rscholar/ipxe/pmagic/pmodules/PMAGIC_2020_12_25.SQFS
- several files on GitHub: https://github.com/search?q=extension%3Asqfs&type=code

---

> In addition I haven't seen such usage.

The `.sfs` extension seems to be also used (though rarely) for SquashFS in some contexts:

- https://github.com/Rprop/libhoudini
- http://wikka.puppylinux.com/squashfs
- https://github.com/puppylinux-woof-CE/woof-CE
- https://bbs.archlinux.org/viewtopic.php?id=232187
- https://wiki.gentoo.org/wiki/How_To_Make_CustomRescueCD

But it's true that the "SFS" abbreviation has **a lot** of different, completely unrelated meanings (and can even refer to another filesystems!): https://wiki.osdev.org/SFS, https://en.wikipedia.org/wiki/Self-certifying_File_System, https://en.wikipedia.org/wiki/Smart_File_System, etc. The _audio format for games_ that you mention is just one of them.

All in all, I'd probably add the `.sqfs` extension instead of `.sfs` (because it's very rare for SquashFS, it seems to be used rather by mistake or out of ignorance):

```suggestion
  file-extension:
    - squashfs
    - snap
    - sqfs
```